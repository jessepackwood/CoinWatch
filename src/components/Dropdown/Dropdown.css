.dropdown {
  height: 20px;
  overflow: hidden;
  display: inline-block;
  width: 125px; }

.dropdown-container {
  border: 1px solid black;
  width: 200px;
  display: inline-block; }

.dropdown-button {
  height: 26px; }
