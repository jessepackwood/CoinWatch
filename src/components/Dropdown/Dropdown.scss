.dropdown {
	height: 35px;
	overflow: hidden;
	display: inline-block;
	width: 180px;
}


select {
	font-size: 18pt;
	background: #fff;
}