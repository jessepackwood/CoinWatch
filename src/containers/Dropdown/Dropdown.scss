.dropdown {
	height: 35px;
	overflow: hidden;
	display: inline-block;
	width: 180px;
}


select {
	font-size: 18pt;
	background: #fff;
  max-height: 400px;
}

optgroup {
  max-height: 400px;
}