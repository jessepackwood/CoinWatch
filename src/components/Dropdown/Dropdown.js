import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import './Dropdown.css'

const Dropdown = ({coins}) => {

  const coinNames = coins.map( (coin, index) => {
    return <option key={`Dropdown - ${index}`}> {coin.long} </option>
  })

  return (
    <select className='dropdown' >
      {coinNames}
    </select>
  )
}

const mapStateToProps = state => ({
  coins: state.coins
})

Dropdown.propTypes = {
  coins: PropTypes.array
}

export default connect(mapStateToProps, null)(Dropdown)
