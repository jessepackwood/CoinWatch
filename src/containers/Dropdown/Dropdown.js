import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import './Dropdown.css'

export const Dropdown = ({coins, onNameChange}) => {
  const coinNames = coins.map( (coin, index) => {
    return <option key={`Dropdown - ${index}`} > {coin.long} </option>
  })

  return (
    <select 
      className='dropdown' 
      onChange={event => onNameChange(event.target.value)}
    >
      <optgroup>
        {coinNames}
      </optgroup>
    </select>
  )
}

export const mapStateToProps = state => ({
  coins: state.coins
})

Dropdown.propTypes = {
  coins: PropTypes.array,
  onNameChange: PropTypes.func
}

export default connect(mapStateToProps, null)(Dropdown)