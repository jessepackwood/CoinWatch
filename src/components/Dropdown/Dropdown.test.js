import Dropdown from './Dropdown'

describe('Dropdown tests', () => {

  it('should be defined', () => {
    expect(Dropdown).toBeDefined()
  })
})