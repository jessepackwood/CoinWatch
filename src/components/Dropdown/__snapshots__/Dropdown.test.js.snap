// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    coins={
        Array [
            "array",
            "of",
            "coins",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <option>
           
           
</option>,
        <option>
           
           
</option>,
        <option>
           
           
</option>,
      ],
      "className": "dropdown",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Dropdown - 0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            undefined,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          undefined,
          " ",
        ],
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "Dropdown - 1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            undefined,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          undefined,
          " ",
        ],
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "Dropdown - 2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            undefined,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          undefined,
          " ",
        ],
        "type": "option",
      },
    ],
    "type": "select",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option>
             
             
</option>,
          <option>
             
             
</option>,
          <option>
             
             
</option>,
        ],
        "className": "dropdown",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Dropdown - 0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              undefined,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            undefined,
            " ",
          ],
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "Dropdown - 1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              undefined,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            undefined,
            " ",
          ],
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "Dropdown - 2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              undefined,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            undefined,
            " ",
          ],
          "type": "option",
        },
      ],
      "type": "select",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
